app/middleware/service-selection.ts

```diff
@@ -1,7 +1,35 @@
 import { NextFunction, Response } from 'express';
 import { ServiceConfig } from '../models/services/base-service';
 import HarmonyRequest from '../models/harmony-request';
-import { chooseServiceConfig } from '../models/services';
+import { chooseServiceConfig, getServiceConfigs } from '../models/services';
+
+/**
+ * Add collections to service configs
+ * @param req - The client request
+ * @returns An array of service configurations
+ */
+function addCollectionsToServicesByAssociation(req: HarmonyRequest): ServiceConfig<unknown>[] {
+  const configs = getServiceConfigs();
+  const { collections } = req;
+  if (collections) {
+    for (const coll of collections) {
+      if (coll.associations?.services) {
+        for (const serviceId of coll.associations?.services) {
+          for (const config of configs) {
+            if (config.umm_s?.includes(serviceId)
+              && config.collections
```

Comment:
Is it ever possible that the service would not have a collections property (i.e. not specified in the YAML file), but _would_ have a umm_s property, in which case, the collections associated with the umm_s record(s) wouldn't get added to the service config here?

Reply:
Argh - I made a commit that changed all this, but forgot to push it. Take a look at the revised code.

Reply:
Ah cool -- I see the added logic.